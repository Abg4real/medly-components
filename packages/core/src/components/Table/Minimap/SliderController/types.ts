export interface Props {
    controllerWidth: number;
}
