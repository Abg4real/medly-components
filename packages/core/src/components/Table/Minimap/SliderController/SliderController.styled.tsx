import { SvgIcon } from '@medly-components/icons';
import { css, styled } from '@medly-components/utils';

const getStyleByState = (state: 'default' | 'hovered' | 'pressed') => css`
    ${SvgIcon} {
        * {
            fill: ${({ theme }) => theme.table.minimap.sliderController.color[state]};
        }
    }
`;

export const SliderController = styled('div')<{ controllerWidth: number }>`
    width: ${({ controllerWidth }) => `${controllerWidth}px`};
    left: 0;
    top: 0;
    cursor: move;
    height: 100%;
    position: absolute;
    display: flex;
    align-items: center;

    ${getStyleByState('default')}

    :hover {
        ${getStyleByState('hovered')}
    }

    :active {
        ${getStyleByState('pressed')}
    }

    ${SvgIcon} {
        position: absolute;
        left: -0.4rem;
    }

    ${SvgIcon}:nth-child(2) {
        left: 1rem;
    }
`;
