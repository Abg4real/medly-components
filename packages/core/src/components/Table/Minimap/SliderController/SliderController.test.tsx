import { cleanup, render } from '@test-utils';
import React from 'react';
import SliderController from '.';

const renderSliderController = () => render(<SliderController controllerWidth={30} />);

describe('Minimap Slider Controller', () => {
    beforeEach(() => {
        cleanup();
    });

    it('should render properly', () => {
        const { container } = renderSliderController();
        expect(container).toMatchSnapshot();
    });
});
