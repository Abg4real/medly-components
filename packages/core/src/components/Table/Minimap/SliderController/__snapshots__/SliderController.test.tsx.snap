// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minimap Slider Controller should render properly 1`] = `
.c2 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c2 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c0 {
  width: 30px;
  left: 0;
  top: 0;
  cursor: move;
  height: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c1 * {
  fill: #126AFA;
}

.c0:hover .c1 * {
  fill: #126AFA;
}

.c0:active .c1 * {
  fill: #126AFA;
}

.c0 .c1 {
  position: absolute;
  left: -0.4rem;
}

.c0 .c1:nth-child(2) {
  left: 1rem;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1 c2"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18.791 11.005H7.621l4.88-4.88c.39-.39.39-1.03 0-1.42a.996.996 0 0 0-1.41 0l-6.59 6.59a.996.996 0 0 0 0 1.41l6.59 6.59a.996.996 0 1 0 1.41-1.41l-4.88-4.88h11.17c.55 0 1-.45 1-1s-.45-1-1-1z"
        fill="#000"
        fill-opacity=".54"
      />
    </svg>
    <svg
      class="c1 c2"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.209 13h11.17l-4.88 4.88c-.39.39-.39 1.03 0 1.42.39.39 1.02.39 1.41 0l6.59-6.59a.996.996 0 0 0 0-1.41l-6.58-6.6a.996.996 0 0 0-1.41 0 .996.996 0 0 0 0 1.41l4.87 4.89H5.209c-.55 0-1 .45-1 1s.45 1 1 1z"
        fill="#000"
        fill-opacity=".54"
      />
    </svg>
  </div>
</div>
`;
